src/main/java/Dude/Dude.java

```diff
@@ -0,0 +1,601 @@
+ * This class handles the conversion of Task objects into a string format suitable for file storage
+ * and the parsing of these strings back into Task objects.
+ */
+class TaskStorage {
+    public static final String FILE_NAME = "tasks.txt";
+
+    /**
+     * Saves a list of tasks to a file, converting each task to a string format.
+     * Each task is written on a new line in the file.
+     *
+     * @param tasks The list of Task objects to be saved.
+     * @throws IOException If an I/O error occurs writing to the file.
+     */
+    public static void saveTasksToFile(ArrayList<Task> tasks) throws IOException {
+        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(FILE_NAME));
+        for (Task task : tasks) {
+            String taskString = taskToFileString(task);
+            if (!taskString.isEmpty()) {
+                bufferedWriter.write(taskString);
+                bufferedWriter.newLine();
+            }
+        }
+
+    }
+
+    /**
+     * Loads a list of tasks from a file, converting each line of the file back into a Task object.
+     *
+     * @return An ArrayList of Task objects read from the file.
+     * @throws IOException If an I/O error occurs reading from the file.
+     */
+    public static ArrayList<Task> loadTasksFromFile() throws IOException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(FILE_NAME));
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            Task task = taskFromString(line);
+            if (task != null) {
+                tasks.add(task);
+            }
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Converts a Task object into a string format for file storage.
+     * The format includes task type, completion status, description, and dates if applicable.
+     *
+     * @param task The Task object to be converted.
+     * @return A string representation of the Task object.
+     */
+    private static String taskToFileString(Task task) {
+        StringBuilder sb = new StringBuilder();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        if (task instanceof ToDo) {
+            sb.append("T|").append(task.isDone ? "1" : "0").append("|").append(task.description);
+        } else if (task instanceof Deadline deadline) {
+            String byDate = deadline.by.format(formatter);
+            sb.append("D|").append(task.isDone ? "1" : "0").append("|").append(task.description).append("|").append(byDate);
+        } else if (task instanceof Event event) {
+            String startDate = event.start.format(formatter);
+            String endDate = event.end.format(formatter);
+            sb.append("E|").append(task.isDone ? "1" : "0").append("|").append(task.description).append("|").append(startDate).append("|").append(endDate);
+        }
+        return sb.toString();
```

Comment:
I like that you use short name for the more locally used variables and long name for more global variables.